hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -213,19 +213,13 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Look up the HostedCluster instance to reconcile
 	hcluster := &hyperv1.HostedCluster{}
-	isMissing := false
 	err := r.Get(ctx, req.NamespacedName, hcluster)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			isMissing = true
```

Comment:
I think we could get in an infinite loop here at if the resource is deleted (NotFound), we return an error, which will requeue the reconcile.  I think we still need to check for `apierrors.IsNotFound(err)` and return with no error in that case.

Reply:
I think you're right — I can instead log and return no error.

Reply:
Fixed, PTAL